e2e-test/e2etest.py

```diff
@@ -0,0 +1,60 @@
+
+from helper import *
+import os
+
+# Setup configuration
+# These values should be injected into the environment before setup
+azure_token = os.environ.get("AZURE_TOKEN")
+bee_name = os.environ.get("BEE_NAME")
+billing_project_name = os.environ.get("BILLING_PROJECT_NAME")
+#TODO: Take in arguments from the command line?
+cbas=False
+wds_upload=True
+cbas_submit_workflow=False
+test_cloning=False
+
+def run_workspace_app_test(cbas, wds_upload, cbas_submit_workflow, test_cloning):
+
+    setup(bee_name)
+    header = {"Authorization": "Bearer " + azure_token};
+
+    # create the workspace
+    workspace_id, workspace_name = create_workspace(cbas, billing_project_name, header)
+
+    # track to see when the workspace WDS is ready to upload data into them
+    # sleep for 5 minutes to allow WDS to start up, if no wds, only sleep 2 minutes to let cbas start up
+    if wds_upload:
+        time.sleep(200)
+    elif cbas_submit_workflow:
+        time.sleep(120)
+    else:
+        print("TEST COMPLETE.")
+
+    if wds_upload:

```

Comment:
Might be more complicated, but adjusting this logic to handle `provisioning` status and letting it poll for progress (and only fail when an error is detected) would make this test more resilient, and also potentially faster to proceed.  Instead of waiting 5 minutes minimum, you could poll every 30 seconds (like the UI does) and proceed ASAP on success.

Reply:
So I am polling in the [get url method](https://github.com/broadinstitute/dsp-reusable-workflows/blob/abf375edc5e2efbe50212a0f866150723c5795d7/e2e-test/helper.py#L77).  I can reduce the initial sleep to some reasonable amount or remove it altogether.  Also definitely need some eyes on the polling logic to make sure it doesn't get stuck there forever